compiler/noirc_evaluator/src/ssa/opt/constant_folding.rs

```diff
@@ -745,15 +745,17 @@ impl ResultCache {
         has_side_effects: bool,
     ) -> Option<CacheResult> {
         self.result.as_ref().and_then(|(origin_block, results)| {
+            if has_side_effects {
+                return None
+            }
```

Comment:
Should this be right at the beginning of the function?

Reply:
Or it could even be done in `get_cached`

Reply:
Yes it can be moved